cmd/mcp-mux/launcher.go

```
@@ -0,0 +1,373 @@
+		if err := startDaemonProcessFrom(launcherPath, enginePath); err != nil {
+			return err
+		}
+		return waitForDaemon(ctlPath, 10*time.Second)
+	}
+
+	lockPath := serverid.DaemonLockPath("", engineName)
+	lock, lockErr := os.OpenFile(lockPath, os.O_CREATE|os.O_WRONLY, 0600)
+	if lockErr == nil {
+		if flockErr := lockFile(lock); flockErr == nil {
+			defer func() {
+				unlockFile(lock)
```

golangci-lint (2.12.2)

[error] Error return value is not checked

(errcheck)